CRM/Upgrade/Incremental/php/FourSeven.php

```diff
@@ -363,4 +373,17 @@ public function deleteVersionCheckCacheFile(CRM_Queue_TaskContext $ctx) {
     return TRUE;
   }
 
+  /**
+   * CRM-17669 and CRM-17686, make scheduled jobs more flexible, disable the 4.6 extension if installed
+   *
+   * @param \CRM_Queue_TaskContext $ctx
+   *
+   * @return bool
+   */
+  public function disableFlexibleJobsExtension(CRM_Queue_TaskContext $ctx) {
+    civicrm_api3('Extension', 'disable', array('key' => 'com.klangsoft.flexiblejobs'));
```

Comment:
`civicrm_api3` will throw an exception if that extension isn't enabled. I'd suggest putting this inside a try/catch (and the catch can just ignore everything).